core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentTaskIndex.java

```diff
@@ -16,30 +16,52 @@
  */
 package org.apache.carbondata.core.datastore.block;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import org.apache.carbondata.core.datastore.BTreeBuilderInfo;
 import org.apache.carbondata.core.datastore.BtreeBuilder;
 import org.apache.carbondata.core.datastore.impl.btree.BlockBTreeBuilder;
+import org.apache.carbondata.core.metadata.AbsoluteTableIdentifier;
 import org.apache.carbondata.core.metadata.blocklet.DataFileFooter;
 
 /**
  * Class which is responsible for loading the b+ tree block. This class will
  * persist all the detail of a table segment
  */
 public class SegmentTaskIndex extends AbstractIndex {
+  private static Map<SegmentKey, SegmentProperties> segmentPropertiesCached =
```

Comment:
I didn't find the class before, I will try to add a segmentproperites in TableSegmentUniqueIdentifier.